File: app/src/test/java/io/github/tobyhs/weatherweight/api/accuweather/forecasts/DailyForecastResponseTest.java
package io.github.tobyhs.weatherweight.api.accuweather.forecasts;

import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.util.List;

import com.google.gson.Gson;
import org.junit.Test;
import org.threeten.bp.format.DateTimeFormatter;

import io.github.tobyhs.weatherweight.AppModule;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class DailyForecastResponseTest {
    @Test
    public void parseJson() throws Exception {
        try (
                InputStream stream = getClass().getResourceAsStream("/accuweather/dailyForecastResponse.json");
                Reader reader = new InputStreamReader(stream)
        ) {
            Gson gson = AppModule.provideGson();
            DailyForecastResponse response = gson.fromJson(reader, DailyForecastResponse.class);

            List<DailyForecast> dailyForecasts =  response.getDailyForecasts();
            assertThat(dailyForecasts.size(), is(1));

            DailyForecast dailyForecast = dailyForecasts.get(0);
            assertThat(dailyForecast.getDate().format(DateTimeFormatter.ISO_LOCAL_DATE), is("2019-01-11"));

            assertThat(dailyForecast.getTemperature().getMinimum().getValue(), is(52.0));
            assertThat(dailyForecast.getTemperature().getMaximum().getValue(), is(57.0));

            assertThat(dailyForecast.getDay().getIconPhrase(), is("Showers"));
            assertThat(dailyForecast.getNight().getIconPhrase(), is("Rain"));
        }
    }
}

File: app/src/test/java/io/github/tobyhs/weatherweight/api/accuweather/forecasts/DailyForecastResponseTest.kt
package io.github.tobyhs.weatherweight.api.accuweather.forecasts

import java.io.InputStreamReader
import java.time.format.DateTimeFormatter

import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.MatcherAssert.assertThat

import org.junit.Test

import io.github.tobyhs.weatherweight.di.AppModule

class DailyForecastResponseTest {
    @Test
    fun parseJson() {
        javaClass.getResourceAsStream("/accuweather/dailyForecastResponse.json").use { stream ->
            InputStreamReader(stream).use { reader ->
                val moshi = AppModule.provideMoshi()
                val moshiAdapter = moshi.adapter(DailyForecastResponse::class.java)
                val dailyForecastResponse = moshiAdapter.fromJson(reader.readText())!!
                val dailyForecasts = dailyForecastResponse.dailyForecasts
                assertThat(dailyForecasts.size, equalTo(1))

                val dailyForecast = dailyForecasts[0]
                assertThat(
                    dailyForecast.date.format(DateTimeFormatter.ISO_LOCAL_DATE),
                    equalTo("2019-01-11")
                )
                assertThat(dailyForecast.temperature.minimum.value, equalTo(52.0))
                assertThat(dailyForecast.temperature.maximum.value, equalTo(57.0))
                assertThat(dailyForecast.day.iconPhrase, equalTo("Showers"))
                assertThat(dailyForecast.day.precipitationProbability, equalTo(80))
                assertThat(dailyForecast.night.iconPhrase, equalTo("Rain"))
                assertThat(dailyForecast.night.precipitationProbability, equalTo(60))
            }
        }
    }
}
